audbackend/core/backend/minio.py

```diff
@@ -0,0 +1,402 @@
+        return checksum
+
+    def _close(
+        self,
+    ):
+        r"""Close connection to repository.
+
+        An error should be raised,
+        if the connection to the backend
+        cannot be closed.
+
+        """
+        # At the moment, this is automatically handled.
+
+    def _collapse(
+        self,
+        path,
+    ):
+        r"""Convert to virtual path.
+
+        <path>
+        ->
+        /<path>
+
+        """
+        path = "/" + path
+        path = path.replace("/", self.sep)
+        return path
+
+    def _copy_file(
+        self,
+        src_path: str,
+        dst_path: str,
+        verbose: bool,
+    ):
+        r"""Copy file on backend."""
+        src_path = self.path(src_path)
+        dst_path = self.path(dst_path)
+        # `copy_object()` has a maximum size limit of 5GB.
+        if self._size(src_path) / 1024 / 1024 / 1024 >= 5:
+            with tempfile.TemporaryDirectory() as tmp_dir:
+                tmp_path = audeer.path(tmp_dir, os.path.basename(src_path))
+                self._get_file(src_path, tmp_path, verbose)
+                checksum = self._checksum(src_path)
+                self._put_file(tmp_path, dst_path, checksum, verbose)
+        else:
+            self._client.copy_object(
+                self.repository,
+                dst_path,
+                minio.commonconfig.CopySource(self.repository, src_path),
+            )
+
+    def _create(
+        self,
+    ):
+        r"""Create repository."""
+        if self._client.bucket_exists(self.repository):
+            utils.raise_file_exists_error(self.repository)
+        self._client.make_bucket(self.repository)
+
+    def _date(
+        self,
+        path: str,
+    ) -> str:
+        r"""Get last modification date of file on backend."""
+        path = self.path(path)
+        date = self._client.stat_object(self.repository, path).last_modified
+        date = utils.date_format(date)
+        return date
+
+    def _delete(
+        self,
+    ):
+        r"""Delete repository and all its content."""
+        # Delete all objects in bucket
+        objects = self._client.list_objects(self.repository, recursive=True)
+        for obj in objects:
+            self._client.remove_object(self.repository, obj.object_name)
+        # Delete bucket
+        self._client.remove_bucket(self.repository)
+
+    def _exists(
+        self,
+        path: str,
+    ) -> bool:
+        r"""Check if file exists on backend."""
+        path = self.path(path)
+        try:
+            self._client.stat_object(self.repository, path)
+        except minio.error.S3Error:
+            return False
+        return True
+
+    def _get_file(
+        self,
+        src_path: str,
+        dst_path: str,
+        verbose: bool,
+    ):
+        r"""Get file from backend."""
+        src_path = self.path(src_path)
+        src_size = self._client.stat_object(self.repository, src_path).size
+        chunk = 4 * 1024
```

Comment:
These are  4 KB. Is this chunk size motivated / tested?  Or ad hoc?

Reply:
I just copied it from the Artifactory backend.

@frankenjoe did you ever tested the influence if the chunk size for downloading data, or did you (or I) copied this from another package?

Reply:
No, I don't think we ever tested a different chunk size.

Reply:
Thanks for the feedback.

I would propose to investigate this at a later stage and independent from this pull request. I created https://github.com/audeering/audbackend/issues/235 to track it.